cdc/owner/changefeed.go

```diff
@@ -298,34 +297,38 @@ func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*
 		return errors.Trace(err)
 	default:
 	}
-	// we need to wait ddl sink to be ready before we do the other things
-	// otherwise, we may cause a nil pointer panic when we try to write to the ddl sink.
-	if !c.sink.isInitialized() {
+	// we need to wait ddl ddlSink to be ready before we do the other things
+	// otherwise, we may cause a nil pointer panic when we try to write to the ddl ddlSink.
+	if !c.ddlSink.isInitialized() {
 		return nil
 	}
-	// This means that the cached DDL has been executed,
-	// and we need to use the latest tables.
-	if c.currentTables == nil {
-		c.currentTables = c.schema.AllTables()
-		log.Debug("changefeed current tables updated",
-			zap.String("namespace", c.id.Namespace),
-			zap.String("changefeed", c.id.ID),
-			zap.Any("tables", c.currentTables),
-		)
+	// TODO: pass table checkpointTs when we support concurrent process ddl
+	allPhysicalTables, minTableBarrierTs, barrier, err := c.ddlManager.tick(ctx, checkpointTs, nil)
+	if err != nil {
+		return errors.Trace(err)
 	}
-	c.sink.emitCheckpointTs(checkpointTs, c.currentTables)
 
-	barrierTs, err := c.handleBarrier(ctx)
+	otherBarrierTs, err := c.handleBarrier(ctx)
 	if err != nil {
 		return errors.Trace(err)
 	}
+	// If there are other barriers less than ddl barrier,
+	// we should wait for them.
+	if barrier.GlobalBarrierTs > otherBarrierTs {
```

Comment:
Better to handle this logic within `handleBarrier `.

Reply:
I will do it in next PR.